kafka-plugins-0.10/src/main/java/io/cdap/plugin/common/KafkaHelpers.java

```diff
@@ -120,4 +121,23 @@ public static void validateKerberosSetting(@Nullable String principal, @Nullable
       throw new IllegalArgumentException(message);
     }
   }
+
+  /**
+   * Validates whether the principal and keytab are both set or both of them are null/empty.
+   * Stores the result in the provided failureCollector.
+   *
+   * @param principal Kerberos principal
+   * @param keytab Kerberos keytab for the principal
+   * @param collector input failureCollector into which the error will be added if present
+   */
+  public static void validateKerberosSetting(@Nullable String principal, @Nullable String keytab,
+      FailureCollector collector) {
+    if (Strings.isNullOrEmpty(principal) != Strings.isNullOrEmpty(keytab)) {
+      String emptyField = Strings.isNullOrEmpty(principal) ? "principal" : "keytab";
+      String message = emptyField + " is empty. When Kerberos security is enabled for Kafka, " +
+          "then both the principal and the keytab have " +
+          "to be specified. If Kerberos is not enabled, then both should be empty.";
+      collector.addFailure(message, null);
```

Comment:
this needs to add a cause so that whatever field(s) have problems are highlighted. Similarly, the message should be specific to what is wrong.

In other words, if the principal is defined and the keytab is not, the keytab should be added as a cause, with a simple message saying the keytab must be specified. If the keytab is defined and the principal is not, the principal should be added as a cause with a message saying the principal must be specified.